### PR TITLE
User options can "yield to" a user option of the same name in superpr…

### PR DESCRIPTION
**Pull request by jpakkane**
…oject. Closes ##2853.

Missing documentation changes. Presented here for your bikeshedding pleasure.

(I did not use `inherit` because inheritance already has a specific meaning in computer science.)

### CODE REVIEW

File: mesonbuild/coredata.py [hunk 1]
```diff
@@ -25,12 +24,20 @@
 version = '0.45.0.dev1'
 backendlist = ['ninja', 'vs', 'vs2010', 'vs2015', 'vs2017', 'xcode']
 
+default_yielding=False
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[Flake8]__

[E225] missing whitespace around operator

File: mesonbuild/optinterpreter.py [hunk 2]
```diff
@@ -83,9 +84,10 @@ def ComboParser(name, description, kwargs):
     for i in choices:
         if not isinstance(i, str):
             raise OptionException('Combo choice elements must be strings.')
-    return coredata.UserComboOption(name, description, choices, kwargs.get('value', choices[0]))
+    return coredata.UserComboOption(name, description, choices, kwargs.get('value', choices[0]),
+    kwargs.get('yielding', coredata.default_yielding),)
```
Comment (ignatenkobrain) [inline mesonbuild/optinterpreter.py]: __[Flake8]__

[E128] continuation line under-indented for visual indent

**Comment by nirbheek** [comment]
Yielding is a good name, I like it. I'll review this on Thursday, travelling right now.

File: test cases/common/174 yield/meson.build [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project('yield_options', 'c')
+
+subproject('sub')
+
+assert(get_option('super_alone') == 'one', 'Unshared option has wrong value in superproject.')
+assert(get_option('super_shared') == 'two', 'Unshared option has wrong value in superproject..')
```
Comment (alyst) [inline test cases/common/174 yield/meson.build]: 'Shared option'?

File: mesonbuild/coredata.py [hunk 4]
```diff
@@ -25,12 +24,20 @@
...
+    def __init__(self, name, description, choices, yielding):
         super().__init__()
         self.name = name
         self.choices = choices
         self.description = description
+        if yielding is None:
+            yielding = default_yielding
+        if not isinstance(yielding, bool):
+            raise RuntimeError('x')
+            raise MesonException('Value of "yielding" must be a boolean.')
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: unreachable

**Comment by ignatenkobrain** [comment]
It seems to be quite late that I read `yelling` instead of `yelding` ;)

code looks good from first glance.

**Comment by jpakkane** [comment]
Any opinions on the default value? I'm liking defaulting to `false`, if only to avoid the tens of bug reports we will probably get about options getting the wrong value from people who will not read the docs.

**Comment by tp-m** [comment]
I can see the rationale for defaulting to `false`.

I was rooting for `true`, but that's merely for convenience. I don't think it will take long for people to add `yield: true` for the relevant options which are in practice probably not too many anyway.

It's going to be a problem when the types are incompatible, no? My main worry is that we're getting ahead of ourselves here and then later people migrate stuff to a built-in more convenient auto option thingy and then the types won't match any more and it will be a bloodbath ;)

File: docs/markdown/Style-guide.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+...
+
+# Style recommendations
+
+This page lists some recommendations on organizing and formatting your
+Meson build files.
+
+## Tabs or spaces?
+
+Always spaces.
```
Comment (TingPing) [inline docs/markdown/Style-guide.md]: Should we recommend 2 spaces as well?

Comment (jpakkane) [reply docs/markdown/Style-guide.md]: I specifically left that out, because I'm a bit torn on whether we should do 2 or 4.

**Pull request merged**